concent_api/core/tests/test_endpoint_protocol_constants.py

```diff
@@ -0,0 +1,78 @@
+    CONCENT_PRIVATE_KEY       = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY        = CONCENT_PUBLIC_KEY,
+    CONCENT_MESSAGING_TIME    = 1,  # seconds
+    FORCE_ACCEPTANCE_TIME     = 2,  # seconds
+    SUBTASK_VERIFICATION_TIME = 3  # seconds
+)
+class ProtocolConstantsTest(ConcentIntegrationTestCase):
+
+    def test_(self):
+        """
+        Tests if call to `provider-constants` endpoint returns protocol constants from settings.
+        """
+
+        protocol_constants_settings = {
+            'CONCENT_MESSAGING_TIME':       settings.CONCENT_MESSAGING_TIME,
+            'FORCE_ACCEPTANCE_TIME':        settings.FORCE_ACCEPTANCE_TIME,
+            'SUBTASK_VERIFICATION_TIME':    settings.SUBTASK_VERIFICATION_TIME,
+            'TOKEN_EXPIRATION_TIME':        settings.TOKEN_EXPIRATION_TIME,
+        }
+
+        response = self.client.get(
+            reverse('core:protocol_constants'),
+        )
+
+        for setting_name, setting_value in protocol_constants_settings.items():
+            self.assertIn(setting_name.lower(),                              response.json())
+            self.assertIn('name',                                            response.json()[setting_name.lower()])
+            self.assertIn('value',                                           response.json()[setting_name.lower()])
+            self.assertEqual(response.json()[setting_name.lower()]['name'],  setting_name)
+            self.assertEqual(response.json()[setting_name.lower()]['value'], setting_value)
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY       = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY        = CONCENT_PUBLIC_KEY,
+)
+class ProtocolConstantsMissingTest(ConcentIntegrationTestCase):
+
+    def test_(self):
+        """
+        Tests if call to `provider-constants` endpoint returns protocol constants from settings
+        with values `undefined` when they are missing.
+        """
+
+        del settings.CONCENT_MESSAGING_TIME
+        del settings.FORCE_ACCEPTANCE_TIME
+        del settings.SUBTASK_VERIFICATION_TIME
+        del settings.TOKEN_EXPIRATION_TIME
+
+        protocol_constants_settings = [
+            'CONCENT_MESSAGING_TIME',
+            'FORCE_ACCEPTANCE_TIME',
+            'SUBTASK_VERIFICATION_TIME',
+            'TOKEN_EXPIRATION_TIME',
+        ]
+
+        response = self.client.get(
+            reverse('core:protocol_constants'),
+        )
+
+        for setting_name in protocol_constants_settings:
+            self.assertIn(setting_name.lower(),                              response.json())
+            self.assertIn('name',                                            response.json()[setting_name.lower()])
+            self.assertIn('value',                                           response.json()[setting_name.lower()])
+            self.assertEqual(response.json()[setting_name.lower()]['name'],  setting_name)
+            self.assertEqual(response.json()[setting_name.lower()]['value'], 'undefined')
```

Comment:
The code should just fail if the critical settings like those timings are undefined. I don't think we should test that case because we don't support it.

Reply:
:+1: